acdc/ioi/utils.py

```diff
@@ -217,34 +224,49 @@ def get_ioi_true_edges(model):
         model = model,
     )
 
-    # remove input -> induction heads connections
-    for layer_idx, head_idx in IOI_CIRCUIT["induction"]:
+    for layer_idx, head_idx in all_nodes:
+        # remove input -> all heads connections
+        # Remove MLP -> all heads connections
         for letter in "qkv":
-            corr.edges[f"blocks.{layer_idx}.hook_{letter}_input"][TorchIndex([None, None, head_idx])][f"blocks.0.hook_resid_pre"][TorchIndex([None])].present = False
-
-    special_connections = {
-        ("s2 inhibition", "name mover"),
-        ("s2 inhibition", "backup name mover"),
-        ("s2 inhibition", "negative"),
-        ("induction", "s2 inhibition"),
-        ("induction"),
-        ("previous token", "induction"),
-        ("duplicate token", "s2 inhibition"),
+            edge_to = corr.edges[f"blocks.{layer_idx}.hook_{letter}_input"][TorchIndex([None, None, head_idx])]
+            edge_to[f"blocks.0.hook_resid_pre"][TorchIndex([None])].present = False
+
+            for mlp_layer_idx in range(layer_idx):
+                edge_to[f"blocks.{mlp_layer_idx}.hook_mlp_out"][TorchIndex([None])].present = False
+
+            for layer_from in range(layer_idx):
+                for head_from in range(12):
+                    edge_to[f"blocks.{layer_from}.attn.hook_result"][TorchIndex([None, None, head_from])].present = False
+
+
+
+    special_connections: set[Conn] = {
```

Comment:
These seem correct, except it seems I never added the output connections to this function?! Perhaps this was conscious due to all the MLP connections, but it seems wrong to me. 

I hope the suggestions work to remove all unnecessary output connections